### PR TITLE
Chemlight Enhancements

### PR DESCRIPTION
**Pull request by voiperr**
***When merged this pull request will:***

**ace_chemlights (new addon):**
- Overhaul chemlight attenuation to avoid overbrightness/spilling and better reflect reality--especially with NVGs: [Vanilla Normal](http://i.imgur.com/zn61eiy.jpg) / [Vanilla NV](http://i.imgur.com/aR0p59M.jpg) vs. [New Normal](http://i.imgur.com/uOkzGwB.jpg) / [New NV](http://i.imgur.com/QqM7iDI.jpg)
- Reconfigure chemlights to last realistic lengths of time (9-12 hours).
- Add orange, white, and [IR](http://i.imgur.com/kg2mOmd.jpg) chemlights.
- Add high-intensity, 30-minute "Hi" chemlight variants (red, yellow, orange, white): [Normal](http://i.imgur.com/zdONM6k.jpg) / [NV](http://i.imgur.com/XMh1W9k.jpg)
- Add [chemlight shields](http://i.imgur.com/QZ6Knp8.png), which can be combined with chemlights to create portable reading lights on the map. Can be disassembled.
- Add "ACE_Box_Chemlights," which contains all chemlight classes and related items.
- Add chemlight snap sound for when thrown.
- Add chemlight snap+shake sound for when attached or inserted into chemlight shield.
- Add action to pickup chemlights off the ground.
- Make chemlight collision noises more subtle (no more clinking).
- References: 
    - https://getcyalume.com/product/cyalume-chemlight-military-grade-chemical-light-sticks
    - https://getcyalume.com/product/military-grade-combat-light-shield-device/
    - http://chem-light.net/militaryusa.php

**ace_attach**:
- Add compat for attaching/detaching IR chemlights.

**ace_flashlights**:
- Remove acc_flashlight configs.
- Config name housekeeping.

**ace_map:**
- Take acc_flashlight config from ace_flashlights (default support for basic flashlight).
- Add option for silent map light activation (e.g. chemlights).
- Make light beam textures sourced from the light item's addon itself.
- Minor config and script housekeeping.

### CODE REVIEW

File: addons/chemlights/XEH_postInit.sqf [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#include "script_component.hpp"
+
+if (isServer) then {
+	[QGVAR(IRTimer), {_this call FUNC(IRTimer)}] call CBA_fnc_addEventHandler;
```
Comment (alganthe) [inline addons/chemlights/XEH_postInit.sqf]: tab here

**Comment by esteldunedain** [comment]
Wow, great job :+1:

**Comment by voiperr** [comment]
Barring more conflicts or git shenanigans, this is ready to go.

**Comment by jonpas** [comment]
This has major line ending issues, overwriting whole files.
(@Glowbal this is why we need #3840)

**Comment by voiperr** [comment]
A rework is on the way.

**Comment by voiperr** [comment]
After discussion with @jonpas, I'm closing this PR and splitting the changes across several branches, for easier testing and evaluation.

**Pull request closed without merging**